### ISSUE TITLE
Rename ComparisonMode.object_id_compare to ComparisonMode.identity_compare?

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#763]
Before the names in the `ComparisonMode` enumeration get set in stone with the Traits 6.0 release, could we consider renaming `ComparisonMode.object_id_compare` to `ComparisonMode.identity_compare`?

My experience with using this enumeration so far is that I never remember what this member is called, and `identity_compare` is the first thing I try. The "object" part seems redundant, and "identity_compare" seems more consistent with the other member names "no_compare" and "equality_compare".

@corranwebster Thoughts?

**Comment by mdickinson** [comment]
Actually, I'm having second thoughts on this. Typical usage is:
```
   foo = MyTrait(comparison_mode=ComparisonMode.identity_compare)
```
and there's a lot of "compare" in that line.

How about if we go even further and just have `ComparisonMode.none`, `ComparisonMode.identity` and `ComparisonMode.equality`?

**Comment by shoeb-github** [comment]
closed via #778.

**Comment by mdickinson** [comment]
Re-opening: the changes in #778 were never merged into master.
